packages/dev/core/src/FlowGraph/flowGraphDataConnection.ts

```diff
@@ -0,0 +1,34 @@
+import type { FlowGraphBlock } from "./flowGraphBlock";
+import { FlowGraphConnection } from "./flowGraphConnection";
+import { FlowGraphConnectionType } from "./flowGraphConnectionType";
+import type { ValueContainer } from "./valueContainer";
+
+/**
+ * @experimental
+ * Represents a connection point for data.
+ * An unconnected input point can have a default value.
+ * An output point will only have a value if it is connected to an input point. Furthermore,
+ * if the point belongs to a "function" node, the node will run its function to update the value.
+ */
+export class FlowGraphDataConnection<T> extends FlowGraphConnection {
+    public constructor(name: string, type: FlowGraphConnectionType, protected _ownerBlock: FlowGraphBlock, private _valueContainer: ValueContainer<T>) {
+        super(name, type);
+    }
+
+    public connectTo(point: FlowGraphDataConnection<T>): void {
+        super.connectTo(point);
+    }

```

Comment:
nit - not needed

Reply:
Actually it is needed to enforce types: we want to make sure the user pass a `FlowGraphDataConnection` and not a `FlowGraphConnection`.

Reply:
Couldn't it be solved with generics at the class level?

Reply:
@RaananW something like [this](https://www.typescriptlang.org/play?#code/C4TwDgpgBAIglgJwgY2HA9gOygXigIjk3ygB8D0BXYfAbgCh6jgIEAzAQ2WgEkBhLJhTAOAIwA20AN70ocqMkHCIAEwAq6AFxR+S1GMlkomSuPEN5UFYmEZM2+ElR2GAX0bJxHAM7eoAIXF0ZABrKBlLMARgiF8YDhEACgBKcNlLeUVMb3RJADoggHNEgHIomN8iQqsEjhLki3l3d3pPHz8AUQAPFGoDCEDgsIgulkwVP0HQtMsR3pYUmYy5LJz8otKVdChvYEo2NgBCesa5ZsYxXYQuYAUvXygBTCF9CQgABXRmAB4nl5Z1Og1FARmMJjo-sJ+gA+KBwAC2YEk8IgmGAfl0zyhbyWckuwGuqAUegBGm0mP+-SMJjMpysNmcWAcDLQWFwBCIdHSeNEVxuUHQAHchAgpiFtGLaFBGJYsv8NIlgNpIahVBo1KkIss4WwoIqABZwbx5axOVnYHCWqDAE0suya7nauTAfXRQXGCDujoIaIIRL4ACCwBYiLQmGqwG2cuE1sF23Quu8HBR9LNdnwDUdlncTpdRry0dVgPZwDplhthdJ2zwee8dPOrXufgAynBCpgOOIVSI3p9mCDRqjwd3+n20d9W+3OyPe180bCtSsSWqtFBJx2uyTR3PbuQaeJ2fu6asCZRUOg-WBKBI4MgBcLWGLtN15v0xQ6nd5KJA-Zns465mQagIEWRcMjgXUDXzU1bDZS08EIYgPydeRazyIURTFPJAOAlIyyaEFxG8aQs21NDKxXAB+bCeiAhY-21HMCPkegWjaB54hEGdJDHYBvjUWFQSHPxuI+Hdvk4jhRN4-joQXR0KMBBxamk8SBOpUwDzwI8FKwK4z0jS9r3EW970woJQglCyQgAGigK8bzvFRam0DVcQyL8fzwx0mLkQoIGASSUlc9zLAgvU0JgxkLStRCSAAMni61DWNRSNGQlCoCQPYEGwcjl0Baj-MC2pvKdVxCOI0KyJS9CH1Fay8nKbg4lKhjMuyyhcuS6Danws4fNYjw9NuCQhkPT0Ams7yTwHeYBmsiavVovo3jFbzWhGnY2w3XieAtD13XXact1nZhEhwlh3ylAB6G77PaKAAFpYXQEJNuyW5vB2zteIAeWoJa1x+zcsVeHidwularushooDuh6HhegV3u+qdxH+6hbTTOCKGoLlZrR3adwB4AACYgeO0HKTOtFEjG0I4YRzg4CI57Xvej7dhqEQ9oOoR3Uk1TzoZ2yoAARiZ+6wEe5G3q525nN5knAe0yahdOiHzsuhahjssmpcRvw5fepWOExm0ovNdl8CoGgGHoInfp3faCxJBUzYtw2WbZk36E9lWK3d9BEgD5h9sNmWkY5x2QYtt2weABUnYxl3MEj2WY7DtFXcrZO48D72OFZ9mUYYIA)? I'm still marking the connectedTo class as abstract so the definition can be strengthened at the subclass level, but the subclasses don't have the connectTo.

Reply:
i thought more adding the generics to the FlowGraphConnection, and setting it here, making those function redefinitions redundant.

Reply:
only if it makes sense. I notice the API has changed since I last reviewed.